app/src/organisms/AppSettings/ConnectRobotSlideout.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react'
+import { useSelector, connect } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+import {
+  Flex,
+  ALIGN_FLEX_END,
+  DIRECTION_COLUMN,
+  SPACING,
+  Text,
+  TYPOGRAPHY,
+  COLORS,
+  Icon,
+  SIZE_2,
+  Link,
+} from '@opentrons/components'
+import { ManualIpHostnameForm } from './ManualIpHostname'
+import { IpHostnameList } from './IpHostnameList'
+import { Slideout } from '../../atoms/Slideout'
+import { PrimaryButton } from '../../atoms/Buttons'
+import { ExternalLink } from '../../atoms/Link/ExternalLink'
+import { Divider } from '../../atoms/structure'
+import { StyledText } from '../../atoms/text'
+import { getScanning, startDiscovery } from '../../redux/discovery'
+import { getConfig } from '../../redux/config'
+
+import type { State } from '../../redux/types'
+import type { MapDispatchToProps } from 'react-redux'
+import type { DiscoveryCandidates } from '../../redux/config/types'
+
+const SUPPORT_PAGE_LINK =
+  'https://support.opentrons.com/en/articles/2934336-manually-adding-a-robot-s-ip-address'
+
+interface SP {
+  candidates: DiscoveryCandidates
+}
+
+interface DP {
+  checkIpAndHostname: () => unknown
+}
+type ConnectRobotSlideoutProps = SP &
+  DP & {
+    isExpanded: boolean
+    onCloseClick: () => unknown
+  }
+
+export function ConnectRobotSlideoutComponent(
+  props: ConnectRobotSlideoutProps
+): JSX.Element | null {
+  const [mostRecentAddition, setMostRecentAddition] = React.useState<
+    string | null
+  >(null)
+  const { onCloseClick, isExpanded } = props
+  const { t } = useTranslation('app_settings')
+  const isScanning = useSelector<State>(getScanning)
+
+  const displayLinkButton = (buttonLable: string): JSX.Element => {
+    return (
+      <Link
+        role="button"
+        css={TYPOGRAPHY.pSemiBold}
+        color={COLORS.blue}
+        onClick={props.checkIpAndHostname}
+        id="AppSettings_Connection_Button"
+      >
+        {buttonLable}
+      </Link>
+    )
+  }
+
+  React.useEffect(() => {
+    if (!isScanning) {
+      setMostRecentAddition(null)
+    }
+  }, [isScanning])
+
+  return (
+    <Slideout
+      title={t('connect_ip')}
+      onCloseClick={onCloseClick}
+      isExpanded={isExpanded}
+      height={`calc(100vh - ${SPACING.spacing4})`}
```

Comment:
subtracting out 1rem was to compensate for the height of the BreadCrumbs bar on deeper than tier 1 pages.  This page is a tier 1 route so you don't need to compensate for the height of the non-existent BreadCrumb bar here.
```suggestion
      height="100%"
```